hyperspy/model.py

```diff
@@ -838,6 +840,67 @@ def _model_function(self, param):
         to_return = self.__call__(non_convolved=False, onlyactive=True)
         return to_return
 
+    def get_raw_signal_for_linear_fitting(self):
+        """Gets the current signal, for linear fitting"""
+        to_return = self.signal()[np.where(self.channel_switches)]
+        if self.signal.metadata.Signal.binned is True:
+            to_return = to_return / self.signal.axes_manager[-1].scale
+        return to_return
+
+    def get_signal_for_linear_fitting(self):
+        """
+        Gets the current signal minus fixed model components, for linear fitting
+        -This could use the previous function to reduce typing-
+        """
+        self._set_p0()
+        deactivated_components = []
```

Comment:
deactivating components should be placed in a try/except/finally statement so that, if something goes wrong, the model doesn't end up in a funny state e.g.

```python

try:
   # Deactive components
   # do something
except:
    raise
finally:
    # reactivate components